packages/web/src/analytics/analytics.ts

```diff
@@ -1,41 +1,49 @@
 import Cookies from 'js-cookie'
 import getConfig from 'next/config'
-const isInEU = require('@segment/in-eu')
 
 let analytics: {
   track: (key: string, properties?: object, options?: object) => void
 }
+
+let analyticsStarted = false
+
 const ALLOW_ANALYTICS_COOKIE_NAME = '__allow__analytics__cookie__'
 const RESPONDED_TO_CONSENT = '__responded_to_consent__'
 
 declare var process: any
 
-export function canTrack(): boolean {
-  return !!Cookies.get(ALLOW_ANALYTICS_COOKIE_NAME) || !isInEU()
+export async function canTrack() {
+  return !!Cookies.get(ALLOW_ANALYTICS_COOKIE_NAME) || !(await isInEU())
 }
 
-export function showVisitorCookieConsent(): boolean {
-  return isInEU() && !Cookies.get(RESPONDED_TO_CONSENT)
+export async function showVisitorCookieConsent() {
+  if (!Cookies.get(RESPONDED_TO_CONSENT)) {
+    return isInEU()
+  }
+  return false
 }
 
-const initializeAnalytics = () => {
-  if (process.browser && canTrack()) {
-    const Segment = require('load-segment')
+async function isInEU() {
+  const inEU = await import('@segment/in-eu').then((mod) => mod.default)
+  return inEU()
+}
+export async function initializeAnalytics() {
+  if (process.browser && !analyticsStarted && (await canTrack())) {
+    const Segment = await import('load-segment').then((mod) => mod.default)
     const { publicRuntimeConfig } = getConfig()
     analytics = Segment({ key: publicRuntimeConfig.__SEGMENT_KEY__ })
   } else {
     analytics = {
       track: () => null,
     }
   }
+  analyticsStarted = true
```

Comment:
Could this possibly be called by multiple functions and do duplicate work? Should you check in progress vs completed?

Reply:
moving this above the import,

its not likely as this should only be called once per "page" when _app.tsx mounts. or when agreement is obtained (which means it wont have been called before)